libr/core/cbin.c

```diff
@@ -215,21 +215,21 @@ R_API bool r_core_bin_strpurge(RCore *core, const char *str, ut64 refaddr) {
 					purge = true;
 					continue;
 				}
+				if (*ptr == '!') {
+					bang = true;
+					ptr++;
+				}
 				range_sep = strchr (ptr, '-');
 				if (range_sep) {
 					*range_sep = 0;
 					from = r_num_get (NULL, ptr);
 					ptr = range_sep + 1;
 					to = r_num_get (NULL, ptr);
 					if (refaddr >= from && refaddr <= to) {
-						purge = true;
+						purge = bang ? false : true;
```

Comment:
You can use just !bang

Reply:
fixed in 7c8252c6f3e5ddcd3a2d3f4a592fb6e2397b4550